packaging/os/yum.py

```diff
@@ -603,6 +603,33 @@ def install(module, items, repoq, yum_basecmd, conf_file, en_repos, dis_repos):
         cmd = yum_basecmd + ['install'] + pkgs
 
         if module.check_mode:
+            ts = yum.rpm.TransactionSet()
+            ts.setVSFlags(yum.rpm._RPMVSF_NOSIGNATURES)
+            yb = yum_base()
+
+            for package in pkgs:
+                if package.startswith('http'):
+                    tmp_file = tempfile.TemporaryFile()
+
+                    try:
+                        tmp_file, info = fetch_url(module, pkg)
+                        tmp_file.write(rsp.read())
```

Comment:
We'll want to chunk reads here like we do at line 532.  rpm files can be rather large and we don't want to read() the whole file into memory at one time.